crypto/mlkem/mlkem-variant.c

```diff
@@ -0,0 +1,1215 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* Copyright (c) 2024, Google Inc. */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <openssl/rand.h>
+#include <openssl/err.h>
+#include <crypto/mlkem.h>
+#include <internal/sha3.h>
+#include <internal/constant_time.h>
+#include <internal/common.h>
+#ifndef NDEBUG
+# include <stdio.h>
+#endif
+
+#ifndef OPENSSL_NO_MLKEM
```

Comment:
I think that this conditioning would be better done in the build.info file.

Reply:
I'll happily remove the guards, especially because this file is not compiled directly, and always included in other files that can be conditionally compiled.